library/openssl/kdf/scrypt_spec.rb

```diff
@@ -0,0 +1,207 @@
+  before :each do
+    @defaults = {
+      salt: "\x00".b * 16,
+      N: 2**14,
+      r: 8,
+      p: 1,
+      length: 32
+    }
+  end
+
+  it "creates the same value with the same input" do
+    key = OpenSSL::KDF.scrypt("secret", **@defaults)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "supports nullbytes embedded into the password" do
+    key = OpenSSL::KDF.scrypt("sec\x00ret".b, **@defaults)
+    key.should == "\xF9\xA4\xA0\xF1p\xF4\xF0\xCAT\xB4v\xEB\r7\x88N\xF7\x15]Ns\xFCwt4a\xC9\xC6\xA7\x13\x81&".b
+  end
+
+  it "coerces the password into a String using #to_str" do
+    pass = mock("pass")
+    pass.should_receive(:to_str).and_return("secret")
+    key = OpenSSL::KDF.scrypt(pass, **@defaults)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "coerces the salt into a String using #to_str" do
+    salt = mock("salt")
+    salt.should_receive(:to_str).and_return("\x00".b * 16)
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, salt: salt)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "coerces the N into an Integer using #to_int" do
+    n = mock("N")
+    n.should_receive(:to_int).and_return(2**14)
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, N: n)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "coerces the r into an Integer using #to_int" do
+    r = mock("r")
+    r.should_receive(:to_int).and_return(8)
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, r: r)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "coerces the p into an Integer using #to_int" do
+    p = mock("p")
+    p.should_receive(:to_int).and_return(1)
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, p: p)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "coerces the length into an Integer using #to_int" do
+    length = mock("length")
+    length.should_receive(:to_int).and_return(32)
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, length: length)
+    key.should == "h\xB2k\xDF]\xDA\xE1.-(\xCF\xAC\x91D\x8F\xC2a\x9C\x9D\x17}\xF2\x84T\xD4)\xC2>\xFE\x93\xE3\xF4".b
+  end
+
+  it "accepts an empty password" do
+    key = OpenSSL::KDF.scrypt("", **@defaults)
+    key.should == "\xAA\xFC\xF5^E\x94v\xFFk\xE6\xF0vR\xE7\x13\xA7\xF5\x15'\x9A\xE4C\x9Dn\x18F_E\xD2\v\e\xB3".b
+  end
+
+  it "accepts an empty salt" do
+    key = OpenSSL::KDF.scrypt("secret", **@defaults, salt: "")
+    key.should == "\x96\xACDl\xCB3/aN\xB0F\x8A#\xD7\x92\xD2O\x1E\v\xBB\xCE\xC0\xAA\xB9\x0F]\xB09\xEA8\xDD\e".b
+  end
+
+  it "accepts an empty length" do
```

Comment:
```suggestion
  it "accepts a zero length" do
```

Reply:
Updated. That does sound like the better description. (I probably copy-pasted the description above, where "empty" makes sense)